gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/handler/HttpGremlinRequestEncoder.java

```diff
@@ -81,7 +90,9 @@ protected void encode(final ChannelHandlerContext channelHandlerContext, final R
             request.headers().add(HttpHeaderNames.CONTENT_TYPE, "application/json");
             request.headers().add(HttpHeaderNames.CONTENT_LENGTH, payload.length);
             request.headers().add(HttpHeaderNames.ACCEPT, serializer.mimeTypesSupported()[0]);
-
+            if (userAgentEnabled) {
+                request.headers().add(UserAgent.USER_AGENT_HEADER_NAME, UserAgent.USER_AGENT);
```

Comment:
Is there reason to not use `HttpHeaderNames.USER_AGENT`?

Reply:
`UserAgent.USER_AGENT_HEADER_NAME` (`"User-Agent"`) matches the header used in websocket requests.
`HttpHeaderNames.USER_AGENT` (`"user-agent"`) may be preferable if we are ok with the inconsistency.

Reply:
As the server header matching is case-insensitive, both of these are interchangeable with gremlin-server. I would agree that it's best to follow the HTTP lower-case convention here.